<a href="https://colab.research.google.com/github/Alej00525/programacion1/blob/main/PARCIAL_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# PARCIAL

## Nombre: Jose Alejandro Parra Nieto
## Codigo: 2260673
## Ti/Edad: 1.095.925.481 / 16
## Cumpleaños: 25/05/09

## PARTE II

In [432]:
import numpy as np
import pandas as pd
from google.colab import files

### Instruccion 1

In [433]:
np.random.seed(0)
datos = np.random.randint(181, 196, 50)

In [434]:
datos # Datos sin filtrar

array([193, 186, 181, 184, 192, 184, 188, 190, 184, 186, 183, 185, 188,
       187, 189, 189, 193, 191, 182, 187, 188, 188, 195, 189, 182, 186,
       190, 194, 189, 190, 185, 184, 181, 184, 186, 195, 181, 183, 184,
       189, 182, 184, 194, 184, 184, 195, 188, 181, 182, 190])

In [435]:
datos[datos > 185] # los datos mayores a 185

array([193, 186, 192, 188, 190, 186, 188, 187, 189, 189, 193, 191, 187,
       188, 188, 195, 189, 186, 190, 194, 189, 190, 186, 195, 189, 194,
       195, 188, 190])

In [436]:
xm = datos.mean()
sd = datos.std()
print(xm) # media de los datos
print(sd) # desviacion estandar de los datos

186.98
4.081617326501836


In [437]:
a = pd.DataFrame({
    'datos':datos
})
print(a) # dataframe de los datos sin filtrar

    datos
0     193
1     186
2     181
3     184
4     192
5     184
6     188
7     190
8     184
9     186
10    183
11    185
12    188
13    187
14    189
15    189
16    193
17    191
18    182
19    187
20    188
21    188
22    195
23    189
24    182
25    186
26    190
27    194
28    189
29    190
30    185
31    184
32    181
33    184
34    186
35    195
36    181
37    183
38    184
39    189
40    182
41    184
42    194
43    184
44    184
45    195
46    188
47    181
48    182
49    190


In [438]:
a_escalada = (a.apply(lambda x:(x-xm)/sd)) # dataframe con los datos escalados
print(a_escalada)

       datos
0   1.474906
1  -0.240101
2  -1.465106
3  -0.730103
4   1.229905
5  -0.730103
6   0.249901
7   0.739903
8  -0.730103
9  -0.240101
10 -0.975104
11 -0.485102
12  0.249901
13  0.004900
14  0.494902
15  0.494902
16  1.474906
17  0.984904
18 -1.220105
19  0.004900
20  0.249901
21  0.249901
22  1.964907
23  0.494902
24 -1.220105
25 -0.240101
26  0.739903
27  1.719906
28  0.494902
29  0.739903
30 -0.485102
31 -0.730103
32 -1.465106
33 -0.730103
34 -0.240101
35  1.964907
36 -1.465106
37 -0.975104
38 -0.730103
39  0.494902
40 -1.220105
41 -0.730103
42  1.719906
43 -0.730103
44 -0.730103
45  1.964907
46  0.249901
47 -1.465106
48 -1.220105
49  0.739903


In [439]:
print(a.describe())

            datos
count   50.000000
mean   186.980000
std      4.123056
min    181.000000
25%    184.000000
50%    186.500000
75%    189.750000
max    195.000000


*analisis obtenido de decribe*
- se tienen 50 datos
- la media de los datos fue 186
- la desviacion estandar fue de 4.12
- el dato mas pequeño fue 181
- el 25% de los datos esta por debajo de 184
- el 50% de los datos esta por debajo de 186
- e- 75% de los datos esta por debajo de  189
- el dato mas grande fue 1.95

In [440]:
a.to_csv('Resultados.csv')
files.download('Resultados.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

### Instruccion 2

In [441]:
df = pd.DataFrame({
    'nombre':['Juan','Ana','Jose','Pepe','Gustavo'],
    'edad':[26,18,23,35,18],
    'calificacion':[4.5,2.9,5,4.3,3.4],
    'ciudad':['Bogota','Medellin','Medellin','Bucaramanga','Bogota']
})
df

,nombre,edad,calificacion,ciudad
0,Juan,26,4.5,Bogota
1,Ana,18,2.9,Medellin
2,Jose,23,5.0,Medellin
3,Pepe,35,4.3,Bucaramanga
4,Gustavo,18,3.4,Bogota


In [442]:
df.head(1)

,nombre,edad,calificacion,ciudad
0,Juan,26,4.5,Bogota


In [443]:
df['nombre']

,nombre
0,Juan
1,Ana
2,Jose
3,Pepe
4,Gustavo


In [444]:
df.loc[df['edad']>18]

,nombre,edad,calificacion,ciudad
0,Juan,26,4.5,Bogota
2,Jose,23,5.0,Medellin
3,Pepe,35,4.3,Bucaramanga


In [445]:
df.assign(edad_10=df['edad'].apply(lambda x: x-10))

,nombre,edad,calificacion,ciudad,edad_10
0,Juan,26,4.5,Bogota,16
1,Ana,18,2.9,Medellin,8
2,Jose,23,5.0,Medellin,13
3,Pepe,35,4.3,Bucaramanga,25
4,Gustavo,18,3.4,Bogota,8


In [446]:
grupo = df.groupby("ciudad")["calificacion"].mean()
grupo

,calificacion
ciudad,
Bogota,3.95
Bucaramanga,4.30
Medellin,3.95


In [447]:
grupo.to_csv('datos procesados.csv')
files.download('datos procesados.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

## PARTE III

In [448]:
PRUEBA_A=np.random.normal(loc=7.4, scale=0.4, size=55)
PRUEBA_B=np.random.gamma(shape=306, scale=0.0229, size=55)

In [449]:
notas = pd.DataFrame({
    'A':PRUEBA_A,
    'B':PRUEBA_B
})
notas

,A,B
0,8.037824,6.416702
1,7.492174,6.834487
2,7.374036,6.579708
3,7.012408,6.587448
4,7.636497,7.412840
5,7.086890,7.351724
6,7.222307,6.909328
7,7.261926,6.786387
8,7.047280,7.791994
9,7.222939,6.811004


In [450]:
print('Analisis prueba A')
print(notas['A'].describe())
print('')
print('Analisis prueba B')
print(notas['B'].describe())

Analisis prueba A
count    55.000000
mean      7.419677
std       0.358890
min       6.706016
25%       7.197590
50%       7.395062
75%       7.613754
max       8.319909
Name: A, dtype: float64

Analisis prueba B
count    55.000000
mean      7.030778
std       0.431510
min       6.176225
25%       6.782515
50%       6.986468
75%       7.396350
max       7.916687
Name: B, dtype: float64


In [451]:
from scipy import stats

In [452]:
print('Prueba de shapiro para la prueba A')
print(stats.shapiro(notas['A']))
print('')
print('Prueba de shapiro para la prueba B')
print(stats.shapiro(notas['B']))

Prueba de shapiro para la prueba A
ShapiroResult(statistic=np.float64(0.9853720314352516), pvalue=np.float64(0.7390740201958141))

Prueba de shapiro para la prueba B
ShapiroResult(statistic=np.float64(0.9774802789309143), pvalue=np.float64(0.38789723492144024))


En la prueba A se dice que estan distribuidos normalmente con una confianza del 73% (valor p > a 0.05)

mientras que en la prueba B se da con una confianza del 38% (valor p > a 0.05)

Es decir, ninguna arroja una confianza del 95%

In [453]:
t_stat, pt_val = stats.ttest_rel(notas["A"], notas['B'])
print("Estadístico t:", t_stat)
print("Valor p:", pt_val)

Estadístico t: 5.047858929044622
Valor p: 5.409316266536659e-06


como el valor p es mayor a 0.05 podemos concluir que los resultados de la prueba no tuvieron diferencias significativas con una confianza del 95% es decir que estan relacionadas

In [454]:
k_stat, pk_val = stats.kendalltau(notas["A"], notas["B"])

print("Coeficiente de correlacion K:", k_stat)
print("Valor p:", pk_val)

Coeficiente de correlacion K: -0.04377104377104378
Valor p: 0.6370195071788382


como el valor p es mayor a 0.05 podemos concluir que los resultados de la prueba no tuvieron diferencias significativas con una confianza del 95% es decir que estan relacionadas

In [455]:
s_stat, ps_val = stats.spearmanr(notas["A"], notas["B"])

print("Coeficiente de correlacion S:", s_stat)
print("Valor p:", ps_val)

Coeficiente de correlacion S: -0.055483405483405496
Valor p: 0.6874375084316097


como el valor p es mayor a 0.05 podemos concluir que los resultados de la prueba no tuvieron diferencias significativas con una confianza del 95% es decir que estan relacionadas

In [456]:
import os
os.makedirs("comparacion", exist_ok=True)

In [457]:
%%writefile comparacion/pruebat.py
import scipy.stats as stats
def prueba_t(a,b):
  t_stat, pt_val = stats.ttest_rel(a, b)
  print("Estadístico t:", t_stat)
  print("Valor p:", pt_val)
  print('')

Overwriting comparacion/pruebat.py


In [458]:
%%writefile comparacion/cor.py
import scipy.stats as stats
def coeficiente_K(a,b):
  k_stat, pk_val = stats.kendalltau(a, b)
  print("Coeficiente de correlacion K:", k_stat)
  print("Valor p:", pk_val)
  print('')
def coeficiente_S(a,b):
  s_stat, ps_val = stats.spearmanr(a, b)
  print("Coeficiente de correlacion S:", s_stat)
  print("Valor p:", ps_val)
  print('')

Overwriting comparacion/cor.py


In [459]:
import importlib
import comparacion.pruebat as pruebat
import comparacion.cor as cor
importlib.reload(pruebat)
importlib.reload(cor)
df = pd.DataFrame({
 "Estudiante": range(1, 11),
 "Prueba_A": [7.1, 7.5, 6.8, 7.3, 7.0, 7.4, 6.9, 7.2, 7.6, 7.1],
 "Prueba_B": [6.9, 7.2, 7.0, 7.1, 6.8, 7.3, 7.0, 7.1, 7.4, 7.0]
})
pruebat.prueba_t(df["Prueba_A"], df["Prueba_B"])
cor.coeficiente_K(df["Prueba_A"], df["Prueba_B"])
cor.coeficiente_S(df["Prueba_A"], df["Prueba_B"])

Estadístico t: 2.282657730758047
Valor p: 0.04835186429465345

Coeficiente de correlacion K: 0.7063224140220168
Valor p: 0.006046856693870933

Coeficiente de correlacion S: 0.8395701571521512
Valor p: 0.002376639566080667

